exporters/otlp/src/otlp_grpc_exporter.cc

```diff
@@ -64,6 +64,27 @@ std::unique_ptr<proto::collector::trace::v1::TraceService::Stub> MakeServiceStub
 {
   std::shared_ptr<grpc::Channel> channel;
 
+  //
+  // Scheme is allowed in OTLP endpoint definition, but is not allowed for creating gRPC channel.
+  // Passing URI with scheme to grpc::CreateChannel could resolve the endpoint to some unexpected
+  // address, so remove schme from the endpoint before passing it to gRPC.
+  //
+  std::string grpc_target;
+  const char *http_scheme  = "http://";
+  const char *https_scheme = "https://";
+  if (options.endpoint.rfind(http_scheme, 0) == 0)
+  {
+    grpc_target = options.endpoint.substr(strlen(http_scheme));
+  }
+  else if (options.endpoint.rfind(https_scheme, 0) == 0)
+  {
+    grpc_target = options.endpoint.substr(strlen(https_scheme));
+  }
+  else
+  {
+    grpc_target = options.endpoint;
+  }
+
```

Comment:
Nit, we can also use [UrlParser](https://github.com/open-telemetry/opentelemetry-cpp/blob/5de793d3cbd312819424d1e6af5fa2a59814bc89/ext/include/opentelemetry/ext/http/common/url_parser.h#L23)?
```
UrlParser url(options.endpoint);
if (url.sucess_) {
  grpc_target = url_.host + ":" + url_.port
}
```
This will ensure that url is in correct format, and remove `path` and `query_string` if added ?